datalad/log.py

```diff
@@ -90,41 +91,80 @@ def format(self, record):
-    log_handler.setFormatter(ColorFormatter())
-    #logging.Formatter('%(asctime)-15s %(levelname)-6s %(message)s'))
-    lgr.addHandler(log_handler)
-    return lgr
-
-lgr = _init_datalad_logger()
-set_level(lgr=lgr)                           # will set the default
+class LoggerHelper(object):
+    """Helper to establish and control a Logger"""
+
+    def __init__(self, name='datalad'):
+        self.name = name
+        self.lgr = logging.getLogger(name)
+
+    def _get_environ(self, var, default=None):
+        return os.environ.get(self.name.upper() + '_%s' % var.upper(), default)
+
+    def set_level(self, level=None, default='WARNING'):
+        """Helper to set loglevel for an arbitrary logger
+
+        By default operates for 'datalad'.
+        TODO: deduce name from upper module name so it could be reused without changes
+        """
+        if level is None:
+            # see if nothing in the environment
+            level = self._get_environ('LOGLEVEL')
+        if level is None:
+            level = default
+
+        try:
+            # it might be a string which still represents an int
+            log_level = int(level)
+        except ValueError:
+            # or a string which corresponds to a constant;)
+            log_level = getattr(logging, level.upper())
+
+        self.lgr.setLevel(log_level)
+
+
+    def get_initialized_logger(self, logtarget=None):
+        """Initialize and return the logger
+
+        Parameters
+        ----------
+        target: string, optional
+          Which log target to request logger for
+        logtarget: { 'stdout', 'stderr', str }, optional
+          Where to direct the logs.  stdout and stderr stand for standard streams.
+          Any other string is considered a filename.  Multiple entries could be
+          specified comma-separated
+
+        Returns
+        -------
+        logging.Logger
+        """
+        # By default mimic previously talkative behavior
+        if logtarget is None:
+            logtarget = self._get_environ('LOGTARGET', 'stdout')
+
+        # Allow for multiple handlers being specified, comma-separated
+        if ',' in logtarget:
+            for handler_ in logtarget.split(','):
+                self.get_initialized_logger(logtarget=handler_)
+            return self.lgr
+
+        if logtarget.lower() in ('stdout', 'stderr') :
+            loghandler = logging.StreamHandler(getattr(sys, logtarget.lower()))
+        else:
+            # must be a simple filename
+            # Use RotatingFileHandler for possible future parametrization to keep
+            # log succinct and rotating
+            loghandler = logging.handlers.RotatingFileHandler(logtarget)
+            # I had decided not to guard this call and just raise exception to go
+            # out happen that specified file location is not writable etc.
+        # But now improve with colors and useful information such as time
+        loghandler.setFormatter(ColorFormatter())
```

Comment:
Maybe we should do this for stream output only. The way it is now, the ColorFormatter causes bad to read file outputs.